internal/git/gitsync_processor.go

```diff
@@ -76,56 +78,107 @@ func watchRepo(ctx context.Context, restConfig *rest.Config, repo *v1.Repository
 	}
 
 	// The revision can be a branch, a tag, or a commit hash
-	h, err := r.ResolveRevision(plumbing.Revision(repo.TargetRevision))
+	hash, err := r.ResolveRevision(plumbing.Revision(repo.TargetRevision))
 	if err != nil {
 		logger.Errorw("error resolving the revision", "revision", repo.TargetRevision, "err", err)
 		return err
 	}
 
-	// Retrieving the commit object matching the hash.
-	commit, err := r.CommitObject(*h)
-	if err != nil {
-		logger.Errorw("error checkout the commit", "hash", h.String(), "err", err)
-		return err
+	namespacedName := types.NamespacedName{
+		Namespace: gitSync.Namespace,
+		Name:      gitSync.Name,
 	}
-
-	// Retrieve the tree from the commit.
-	tree, err := commit.Tree()
-	if err != nil {
-		logger.Errorw("error get the commit tree", "err", err)
-		return err
+	gitSync = &v1.GitSync{}
+	if err = k8Client.Get(ctx, namespacedName, gitSync); err != nil {
+		// if we aren't able to do a Get, then either it's been deleted in the past, or something else went wrong
+		if apierrors.IsNotFound(err) {
+			return nil
+		} else {
+			logger.Errorw("Unable to get GitSync", "err", err)
+			return err
+		}
 	}
+	val, ok := gitSync.Status.CommitStatus[repo.Name]
+	// Only create the resources for the first time if not created yet.
+	// Otherwise, monitoring with intervals.
+	if !ok {
+		// Retrieving the commit object matching the hash.
+		commit, err := r.CommitObject(*hash)
+		if err != nil {
+			logger.Errorw("error checkout the commit", "hash", hash.String(), "err", err)
+			return err
+		}
 
-	if !isRootDir(repo.Path) {
-		// Locate the tree with the given path.
-		tree, err = tree.Tree(repo.Path)
+		// Retrieve the tree from the commit.
+		tree, err := commit.Tree()
 		if err != nil {
-			logger.Errorw("error locate the path", "err", err)
+			logger.Errorw("error get the commit tree", "err", err)
 			return err
 		}
-	}
 
-	// Read all the files under the path and apply each one respectively.
-	err = tree.Files().ForEach(func(f *object.File) error {
-		logger.Debugw("read file", "file_name", f.Name)
-		//TODO: this currently assumes that one file contains just one manifest - modify for multiple
-		manifest, err := f.Contents()
+		if !isRootDir(repo.Path) {
+			// Locate the tree with the given path.
+			tree, err = tree.Tree(repo.Path)
+			if err != nil {
+				logger.Errorw("error locate the path", "repository path", repo.Path, "err", err)
+				return err
+			}
+		}
```

Comment:
To make @shubhamdixit863 's job of merging a little easier, can you add this same `getCommitTreeAtPath()` method from his [PR](https://github.com/numaproj-labs/numaplane/pull/47/files) and call it here?

Reply:
yeah, sure, but would prefer to do it in a follow up PR. I can even do it after his PR is merged. That would be easier for all of us.

Reply:
I think if yours is merged, then he will need to do it, but maybe that's okay